streaming/Streaming.mli

```diff
@@ -153,6 +153,10 @@ module Source : sig
   val filter : ('a -> bool) -> 'a t -> 'a t
   (** A source that includes only the elements that satisfy a predicate. *)
 
+  val filter_map : ('a -> 'b option) -> 'a t -> 'b t
+  (** Filter_map f source applies f to every element x of source, discard it if
+   * f x is None, and keeps y if f x = Some y *)
```

Comment:
A few points about comments:

- I suggest wrapping code into `[...]` (see the `zip` function above, for example).
- `Filter_map` should be `filter_map`.
- Docstrings are currently not supposed to have leading `*` (see https://github.com/ocaml/odoc/issues/352).

Essentially, I would suggest to use this for the comment:

```ocaml
(** [filter_map f source] applies [f] to every element [x] of source, discarding it if
    [f x] produces [None], and keeping the transformed value otherwise.  *)
```